tests/test_singleton.py

```diff
@@ -0,0 +1,30 @@
+import pytest
+
+import singleton
+
+
+def test_singleton(singleton_object_1, singleton_object_2):
+    assert singleton_object_1 == singleton_object_2
+
+    assert singleton_object_1.value == singleton_object_2.value
+    singleton_object_1.value = 10
+    assert singleton_object_2.value == 10
+
+
+@pytest.fixture
+def singleton_test_class():
+    class TestSingleton(metaclass=singleton.Singleton):
```

Comment:
This way for every call of this fixture you'll be having new class object. And this is strange for me at first sight why these tests pass.